src/collection-view.js

```diff
@@ -567,21 +505,36 @@ const CollectionView = Backbone.View.extend({
     this.Dom.detachEl(view.el, view.$el);
   },
 
-  _renderChildren(views) {
-    if (this.isEmpty(!views.length)) {
+  _renderChildren() {
+    const views = this._addedViews || this.children._views;
+
+    this.triggerMethod('before:render:children', this, views);
+
+    if (this.isEmpty()) {
```

Comment:
one major change here is that the `render:children` events will now fire even if the empty view is shown.  Previously with the two `isEmpty` checks this was fairly difficult without artificially triggering those events in odd cases.  Now we can throw the event when 0 children are rendered.  This will simplify having an event for when the children are rendered even if there is no empty view.